src/05-many-two-state-systems.md

```diff
@@ -0,0 +1,768 @@
+
+```python
+H = A*J[2] + delta*J[0]
+J2 = J[0]*J[0] + J[1]*J[1] + J[2]*J[2]
+commutator(H,J2)
+```
+
+The zero commutator of $J^2$ with the Hamiltonian means something more - it means that angular momentum is conserved over time, i.e. if we start in a state of a particular $j$ then we can't move to a new $j$. This explains the flat line in Fig 4. We started the simulation in energy level 0 (aka $|1, -1 \rangle$) but the system is incapable of going level 2 (aka $|0, 0 \rangle$) because it has a different angular momentum $j$.
+
+We can confirm these ideas by starting the system off in level 2 instead of level 0:
+
+```python
+result, ekets = simulate(A, delta, J, times, 2)
+```
+
+```python
+psi, P = make_p_psi_arrays(result.states, ekets) 
+```
+
+```python
+df = pd.DataFrame(P, index=times)
+```
+
+```python
+df.plot(figsize=(10,8), 
+        title = "2 TSS,   $H =A \ J_z + \delta \ J_x \  \cos (\omega t)$     (A=0.1, $\omega = 0.1$, $\delta=0.001$)   (Fig 5)")
+plt.xlabel("Time")
+plt.ylabel("Probability");
+```
+
+Fig 5 shows that the system doesn't change over time. This is because there is only 1 state with $j=0$, namely $|0,0\rangle$, and so there is nowhere for the state to evolve to without violating the conservation of "angular momentum". 
+
+We can now also solve the mystery of why level 1 in fig 1 appears to different to the others - it's because it has a different angular "angular momentum" ($j=0$ for level 1, $j=1$ for the others).
+
+> I use quotation marks around "angular momentum" to remind us that it's not really angular momentum, but something that is mathematically equivalent. Some people call $j$ the Dicke cooperation number to remind us of this distinction.
+
+Conservation of $j$ allows to hugely reduce the computationally resources required for simulating many TSS. Let's see how this works before we finish up.
+
+
+# 5.7 - Isolated universes
+
+
+When we began this tutorial, we described the states of 2 TSS in the |±, ±> basis because it felt natural. For $N$ TSS, such a description creates a very large number of states to consider - specifically $2^{N}$.
+
+Now that we have learnt about the conservation of $j$, not all of the $2^{N}$ states need to be considered in every simulation. Each value of $j$ lives in its own universe (much like [parity in tutorial 4](https://nbviewer.jupyter.org/github/project-ida/two-state-quantum-systems/blob/master/04-spin-boson-model.ipynb#4.4---Parity)) and is disconnected from the others. The largest number of states that we need to consider comes when we have maximum $j$, i.e. $j_{\text{max}} = \frac{1}{2} N$. In this case, the number of possible $m$ values is $2j+1 = N+1$ - this grows linear with $N$ rather than exponentially so is much more favourable for doing simulations for large $N$.
```

Comment:
I would just mention that they are disconnected because the commutator [J2, H] =0